common/rule-processor/src/RuleHelper.js

```diff
@@ -97,18 +97,27 @@ class RuleMatcher {
    * In some cases like wildcard match or regex match, resultingUrl will be destination+replaced group variables.
    */
   static matchUrlWithRuleSource(sourceObject, url, destination) {
+    return RuleMatcher.matchUrlWithRuleSourceWithExtraInfo(sourceObject, url, destination).destination;
+  }
+
+  static matchUrlWithRuleSourceWithExtraInfo(sourceObject, url, destination) {
     var operator = sourceObject.operator,
       urlComponent = extractUrlComponent(url, sourceObject.key),
       value = sourceObject.value,
       blackListedDomains = CONSTANTS.BLACK_LIST_DOMAINS || [];
 
     for (var index = 0; index < blackListedDomains.length; index++) {
       if (url.indexOf(blackListedDomains[index]) !== -1) {
-        return null;
+        return { destination: null };
       }
     }
-
-    return RuleMatcher.matchUrlCriteria(urlComponent, operator, value, destination);
+    const result = RuleMatcher.matchUrlCriteria(urlComponent, operator, value, destination);
+    if (result) {
+      if (typeof result === "string") return { destination: result };
+      else return { destination: result.destination, matches: result.matches };
+    } else {
+      return { destination: null };
+    }
```

Comment:
This can be removed and moved to `matchUrlCriteria` instead where we have switch cases for this